classes/Configuration.php

```diff
@@ -601,13 +613,26 @@ public static function deleteFromContext($key, int $idShopGroup = null, int $idS
             $idShop = Shop::getContextShopID(true);
         }
 
-        $id = Configuration::getIdByName($key, $idShopGroup, $idShop);
+        $configurationId = Configuration::getIdByName($key, $idShopGroup, $idShop);
+
+        self::deleteMultiShopConfigurationByKey($key, $configurationId, $idShopGroup, $idShop);
+    }
+
+    /**
+     * @param mixed $key
+     * @param int|null $configurationId
+     * @param int|null $idShopGroup
+     * @param int|null $idShop
+     */
+    public static function deleteMultiShopConfigurationByKey($key, ?int $configurationId, ?int $idShopGroup, ?int $idShop): void
```

Comment:
```suggestion
    public static function deleteFromGivenContext($key, ?int $configurationId, ?int $idShopGroup, ?int $idShop): void
```

To match the naming of `getIdByNameFromGivenContext`

Reply:
But it's weird to have both the key and the configuration ID, once you have the ID you don't the key nor the shopId/shopGroupId And in the other use case you don't provide the ID so you need to handle a fallback

You should split this method into two again:
- `public static function deleteFromGivenContext(string $key, ?int $idShopGroup, ?int $idShop): void`
- `public static function deleteById(int $configurationId): void`

So that `deleteFromGivenContext` and `deleteFromContext` will rely on `deleteById`

Reply:
nice catch 👍